File: macos/Tests/ElevateCoreTests/PolicyNotesTests.swift
import Testing
import Foundation
@testable import ElevateCore

struct PolicyNotesTests {
    private func policy(approval: Bool = false, mfa: Bool = false, context: String? = nil) -> RolePolicy {
        RolePolicy(defaultDuration: .seconds(3600), maximumDuration: .seconds(28800), requiresJustification: true,
                   requiresTicket: false, requiresMFA: mfa, requiresApproval: approval, authenticationContext: context)
    }

    @Test func plainPolicyHasNoNotes() {
        #expect(PolicyNotes.labels(for: policy()).isEmpty)
        #expect(PolicyNotes.caption(for: policy()) == nil)
        #expect(PolicyNotes.explanation(for: policy()) == nil)
        #expect(PolicyNotes.actionTitle(for: policy()) == "Activate")
    }

    @Test func labelsAreOrderedApprovalThenStepUps() {
        let p = policy(approval: true, mfa: true, context: "c1")
        #expect(PolicyNotes.labels(for: p) == ["approval", "MFA", "Conditional Access"])
        #expect(PolicyNotes.caption(for: p) == "approval · MFA · Conditional Access")
        #expect(PolicyNotes.actionTitle(for: p) == "Request")
    }

    @Test func conditionalAccessExplanationNamesTheContext() {
        let text = PolicyNotes.explanation(for: policy(context: "c1"))
        #expect(text?.contains("authentication context c1") == true)
        #expect(text?.contains("approver") == false)
    }
}

File: windows/tests/Elevate.Core.Tests/PolicyNotesTests.cs
using Elevate.Core.Models;
using Elevate.Core.Support;
using FluentAssertions;

namespace Elevate.Core.Tests;

public class PolicyNotesTests
{
    private static RolePolicy Policy(bool approval = false, bool mfa = false, string? context = null) => new(
        TimeSpan.FromHours(1), TimeSpan.FromHours(8), RequiresJustification: true, RequiresTicket: false,
        RequiresMfa: mfa, RequiresApproval: approval, AuthenticationContext: context);

    [Fact]
    public void PlainPolicyHasNoNotes()
    {
        PolicyNotes.Labels(Policy()).Should().BeEmpty();
        PolicyNotes.Caption(Policy()).Should().BeNull();
        PolicyNotes.Explanation(Policy()).Should().BeNull();
        PolicyNotes.ActionTitle(Policy()).Should().Be("Activate");
    }

    [Fact]
    public void LabelsAreOrderedApprovalThenStepUps()
    {
        var policy = Policy(approval: true, mfa: true, context: "c1");

        PolicyNotes.Labels(policy).Should().Equal("approval", "MFA", "Conditional Access");
        PolicyNotes.Caption(policy).Should().Be("approval · MFA · Conditional Access");
        PolicyNotes.ActionTitle(policy).Should().Be("Request");
    }

    [Fact]
    public void ConditionalAccessExplanationNamesTheContext()
    {
        var text = PolicyNotes.Explanation(Policy(context: "c1"));

        text.Should().Contain("authentication context c1");
        text.Should().NotContain("approver");
    }
}
